docs/user_guide/high_availabilty.rst

```diff
@@ -0,0 +1,197 @@
+      new unique SSID to it.
+    - If there is no other SP available after the hot SP becomes offline, the overseer starts reporting no hot
+      SP and no SSID.
+    - During the above case, when either a new SP or the previous hot SP comes online, the overseer will create a new
+      unique SSID and assign that SP as hot.
+    - When the hot SP is keeping its online state by maintaining communication with the overseer, newly joined SPs are
+      marked as online, but not hot SP.  Newly joined SPs also have no SSIDs associated with them, which means the
+      original SSID is maintained with the current hot SP.
+    - The overseer will include information on all joined SP and hot SP information in the reply to heartbeats.
+
+Overseer Agent
+==============
+Overseer Agents provide a convenient and unified interface to communicate with the Overseer. By
+following the same interface and behavior defined by the the Overseer Agent specification, users can implement their own
+Overseer and Overseer Agent as a drop-in replacement.
+
+The Overseer Agent periodically sends heartbeats to the Overseer. These heartbeats indicate that this Overseer Agent is still
+alive and the communication between Overseer and Overseer Agent is not broken. When it fails in sending heartbeats,
+the Overseer Agent shall retry until it is told to stop.
+
+Each Overseer Agent includes a special property called "role". This property is set by the callers as the callers, namely FL
+servers, FL clients or FL admins, have the knowledge about the role. For example:
+
+    - The Overseer Agents used by FL servers have to tell the Overseer that it sends heartbeats on behalf of an SP, thus
+      its role is "server."
+    - The Overseer Agents used by admin must indicate its role so Overseers allow admin commands after the agents are
+      authenticated and authorized.
+
+The Overseer Agents internally maintain the last reply from Overseers, which is received during the last heartbeat.
+
+Callers can register a callback. The execution of the callback is based on the flag ``conditional_cb``.
+
+    - If the callback is registered with the conditional_cb flag set to True, the callback is called only when the SSID changes.
+    - If that flag is set to False, the callback is called immediately every time after the reply of one heartbeat from the
+      overseer is received.
+
+Overseer Response Handling
+==========================
+The Overseer client is responsible for handling Overseer responses (or the lack of responses) properly.
+
+FL Client
+---------
+No response from Overseer (connection error, etc.)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+I’ll keep retrying to communicate with the Overseer. If I already have a hot SP, I’ll keep using it.
+
+No hot SP available
+^^^^^^^^^^^^^^^^^^^
+I’ll keep retrying to communicate with the Overseer. If I already have a hot SP, I’ll keep using it.
```

Comment:
The "No hot SP available" behavior is the same as previous "No response from Overseer".   I think this should state that the client will continue retrying to communicate with the current hot SP, while also continuing to communicate with the overseer and responding to any hot SP change indicated in the overseer response.